sklearn/ensemble/_hist_gradient_boosting/binning.py

```diff
@@ -13,89 +13,74 @@
 from ...base import BaseEstimator, TransformerMixin
 from ...utils.validation import check_is_fitted
 from ._binning import _map_to_bins
-from .common import X_DTYPE, X_BINNED_DTYPE, ALMOST_INF
+from .common import X_DTYPE, X_BINNED_DTYPE, ALMOST_INF, X_BITSET_INNER_DTYPE
+from ._bitset import set_bitset_memoryview
 
 
-def _find_binning_thresholds(data, max_bins, subsample, random_state):
-    """Extract feature-wise quantiles from numerical data.
+def _find_binning_thresholds(col_data, max_bins):
```

Comment:
Note for reviewers: This is mostly unchanged, we just made it work on single columns instead of the whole data